### ISSUE TITLE
Creation of a 0 interval time

### ISSUE THREAD

**Issue by sitawit** [briannesbitt/Carbon#369]
### Suggest
To be able to create a time interval of **0**.

So the following code would work.
```php
CarbonInterval::months(0);   // 0 months
CarbonInterval::hours(0);    // 0 hours
```

### Current version
So far, CarbonInterval can't create a 0 interval of any period.

For example:
```php
CarbonInterval::months(0); 
CarbonInterval::hours(0); 
```

Will result in 

```php
Exception
DateInterval::__construct() [<a href='0dateinterval.construct0'>dateinterval.construct0</a>]: Unknown or bad format (P)
```

### Background
At first, it seems make no sense to create a 0 interval because in fact, it doesn't contain any interval. However, when it comes to a situation where you create a set of time intervals. And the first slot start from 00:00. So in this case, you have to create this 0 interval using other method rather than CarbonInterval. In order to make your code consistency, using CarbonInterval for all the time in the set would be great. And you don't have to create a logic where it catches an exception of 0 interval.

**Comment by dereuromark** [comment]
Indeed: Sounds useful to support that. Especially with dynamic data (>= 0) coming in.

**Comment by Sverri** [comment]
When the constructor checks the parameter it uses the expression, `$months > 0`. It looks like this is just a simple mistake that can be fixed by simply changing it to `$months > -1`. PHP's  `DateInterval` allows a "zero interval", i.e. something like `P0M`, so it should not be a problem (unless I am missing something about how `CarbonInterval` works).

**Comment by jarnovanleeuwen** [comment]
https://github.com/briannesbitt/Carbon/pull/437 should fix this.

**Comment by briannesbitt** [comment]
Merged #437
